tools/retrive_go_repos.py

```diff
@@ -0,0 +1,302 @@
+        if str_line.startswith('name = '):
+            splited_text = str_line.split("=")
+            if len(splited_text) > 1:
+                pkg = splited_text[1].strip().replace('"', '')
+                get_go_pkg_github_repo_details(pkg)
+
+
+def get_dependancy_data_from_vendor_folder(org_repo: str):
+    """Get Dependancy repo details from vendor folder."""
+    sha = get_commit_sha(org_repo)
+    vendor_folder_git_url = get_vendor_folder_git_tree_url(org_repo, sha)
+    if vendor_folder_git_url:
+        get_dependancy_repo_from_vendor(vendor_folder_git_url, 0, "")
+    else:
+        logging.error("Unble to find dependancy file (go.mod/Gopkg.lock) or Vendor folder for '{org_repo}'"
+                      .format(org_repo=org_repo))
+
+
+def get_dependancy_data(org_repo: str):
+    """
+    Get Dependancy repo details.
+
+    After anlysing few repos, we came up with below logic to get dependant github repos for a repo.
+
+    if go.mod file present in the repo:
+        then we are taking dependant go packages mentioned into require space.
+    else if Gopkg.lock file present in the repo:
+        then we are taking all the dependant packages mentioned into that file.
+    else if If Vendor folder present inside repo:
+        then we are taking dependency packages based on sub-folder structure.
+
+    Note:  Tried to use asyncio for parallel execution but we are facing issue with pkg.go.dev url,
+    as its limitting no of request,  so removed that code.
+    """
+    try:
+        get_dependancy_data_from_go_mod_file(org_repo)
+    except URLError:
+        try:
+            get_dependancy_data_from_lock_file(org_repo)
+        except URLError:
+            get_dependancy_data_from_vendor_folder(org_repo)
+
+
+def get_commit_sha(org_repo: str):
+    """Get commit sha for a github repo."""
+    commit_raw_url = "https://api.github.com/repos/{org_repo}/commits/master"
+    url = commit_raw_url.format(org_repo=org_repo)
+    try:
+        result = requests.get(url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+        if result.status_code == 200:
+            return result.json()['sha']
+        else:
+            logging.error("repo doesn't exist : " + org_repo)
+    except Exception as ex:
+        logging.error("Error while retriving commit details for repo : {repo}, msg : {msg}"
+                      .format(repo=org_repo, msg=str(ex)))
+    return None
+
+
+def get_vendor_folder_git_tree_url(org_repo: str, sha: str):
+    """Get vendor folder git tree url."""
+    repo_structure_raw_url = "https://api.github.com/repos/{org_repo}/git/trees/{sha}"
+    url = repo_structure_raw_url.format(org_repo=org_repo, sha=sha)
+
+    try:
+        result = requests.get(url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+        if result.status_code == 200:
+            vendor_folder_info = list(filter(lambda x: (x['path'] == "vendor"), result.json()['tree']))
```

Comment:
Shouldn't this be checking `if "vendor" in x['path']` or are we only considering first level dependencies?

Reply:
If only first level you can directly check if one of the strings is vendor, since there will be only one such folder

Reply:
> vendor

In the path we will get only "vendor", not the entire url, so condition seems correct.

Reply:
> If only first level you can directly check if one of the strings is vendor, since there will be only one such folder

Later on I am using vendor_folder_info to retrive the url, so filtering it and checking length  and later on using same poperty to get url.